services/slam/client_test.go

```diff
@@ -350,6 +378,12 @@ func TestFailingClient(t *testing.T) {
 		test.That(t, err.Error(), test.ShouldContainSubstring, "failure to get position")
 		test.That(t, p, test.ShouldBeNil)
 
+		// test get position
+		pose, componentRef, err := failingSLAMClient.GetPosition(context.Background(), nameFail)
+		test.That(t, err.Error(), test.ShouldContainSubstring, "failure to get position")
+		test.That(t, pose, test.ShouldBeNil)
```

Comment:
[q] how is pose nil here when you return poseFail up above?

Reply:
Because this is the client test. The client does not return pose on an error.